Signal/src/views/RemoteVideoView.m

```diff
@@ -0,0 +1,219 @@
+                    [subview autoPinEdgesToSuperviewEdges];
+                } else {
+                    OWSFail(@"New subviews added to MTLVideoView. Reconsider this hack.");
+                }
+            }
+        }
+    }
+#elif defined(__arm64__)
+    // Canary incase the upstream RTC_SUPPORTS_METAL macro changes semantics
+    OWSFail(@"should only use legacy video view on 32bit systems");
+#endif
+
+    // On 32-bit iOS9+ systems, use the legacy EAGL backed view.
+    if (_videoRenderer == nil) {
+        RTCEAGLVideoView *eaglVideoView = [RTCEAGLVideoView new];
+        eaglVideoView.delegate = self;
+        _videoRenderer = eaglVideoView;
+        [self addSubview:_videoRenderer];
+        // Pinning legacy RTCEAGL view discards aspect ratio.
+        // So we have a more verbose layout in the RTCEAGLVideoViewDelegate methods
+        // [_videoRenderer autoPinEdgesToSuperviewEdges];
+    }
+
+    // We want the rendered video to go edge-to-edge.
+    _videoRenderer.layoutMargins = UIEdgeInsetsZero;
+
+    return self;
+}
+
+#pragma mark - RTCVideoRenderer
+
+/** The size of the frame. */
+- (void)setSize:(CGSize)size
+{
+    [self.videoRenderer setSize:size];
+}
+
+#pragma mark - RTCEAGLVideoViewDelegate
+
+- (void)videoView:(RTCEAGLVideoView *)videoView didChangeVideoSize:(CGSize)remoteVideoSize
+{
+    AssertIsOnMainThread();
+    CGFloat aspectRatio = remoteVideoSize.width / remoteVideoSize.height;
+
+    DDLogVerbose(@"%@ Remote video size: width: %f height: %f ratio: %f",
+        self.logTag,
+        remoteVideoSize.width,
+        remoteVideoSize.height,
+        aspectRatio);
+
+    UIView *containingView = self.superview;
+    if (containingView == nil) {
+        DDLogDebug(@"%@ Cannot layout video view without superview", self.logTag);
+        return;
+    }
+
+    if (![self.videoRenderer isKindOfClass:[RTCEAGLVideoView class]]) {
+        OWSFail(@"%@ Unexpected video renderer: %@", self.logTag, self.videoRenderer);
+        return;
+    }
+
+    [NSLayoutConstraint deactivateConstraints:self.remoteVideoConstraints];
+
+    NSMutableArray<NSLayoutConstraint *> *constraints = [NSMutableArray new];
+    if (remoteVideoSize.width > 0 && remoteVideoSize.height > 0 && containingView.bounds.size.width > 0
+        && containingView.bounds.size.height > 0) {
+
+        [constraints addObject:[videoView autoPinToAspectRatio:aspectRatio]];
+        [constraints addObject:[videoView autoSetDimension:ALDimensionWidth
+                                                    toSize:containingView.width
+                                                  relation:NSLayoutRelationGreaterThanOrEqual]];
+        [constraints addObject:[videoView autoSetDimension:ALDimensionHeight
+                                                    toSize:containingView.height
+                                                  relation:NSLayoutRelationGreaterThanOrEqual]];
+        [constraints addObjectsFromArray:[videoView autoCenterInSuperview]];
```

Comment:
What's to guarantee that `videoView` will have the smallest possible size that satisfies these constraints?

Reply:
Done - added low priority "pin to superview edges" constraint.